assets/js/aplus.js

```diff
@@ -584,65 +588,182 @@ $.fn.highlightCode = function(options) {
+        catch(e) {
+            return false;
+        }
+      },
+
+      filterGetType: function(query) {
+          if (!query) {
+            // Empty query, no filtering
+            return [undefined, ""];
+          }
+
+          if (query.startsWith("<")) {
+            if (query.startsWith("<=")) {
+                return [this.lteFilter, query.slice(2)];
+            }
+            else {
+                return [this.ltFilter, query.slice(1)];
+            }
+          }
+
+          else if (query.startsWith(">")) {
+            if (query.startsWith(">=")) {
+                return [this.gteFilter,  query.slice(2)];
+            }
+            else {
+                return [this.gtFilter, query.slice(1)];
+            }
+          }
+
+          else if (query.startsWith("==")) {
+              return [this.eFilter, query.slice(2)];
+          }
+
+          else if (query.startsWith("!=")) {
+              return [this.neFilter,  query.slice(2)];
           }
-          this.element.find('thead').append(filterRow);
-        },
 
-        filterColumn: function(input) {
+          else if (query.match(/\//g) && query.match(/\//g).length >= 2) {
+            if (this.validateRegex(query)) {
+              return [this.regexFilter, query];                      
+            }
+            else {
+              return [this.normalFilter, query];
+            }
+          }
+
+          else {
+            return [this.normalFilter, query];
+          }
+      },
+
+      lteFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text <= query : 
+                                              parseFloat(text) <= parseFloat(query);
+      },
+
+      ltFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text < query : 
+                                            parseFloat(text) < parseFloat(query);
+      },
+
+      gteFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text >= query : 
+                                            parseFloat(text) >= parseFloat(query);
+      },
+
+      gtFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text > query : 
+                                            parseFloat(text) > parseFloat(query);
+      },
+
+      eFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text == query : 
+                                            parseFloat(text) == parseFloat(query);
+      },
+
+      neFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text != query : 
+                                            parseFloat(text) != parseFloat(query);
+      },
+
+      regexFilter: function(text, query) {
+        var parts = query.split('/');
+        var regex = query;
+        var options = "";
+        if (parts.length > 1) {
+            regex = parts[1];
+            options = parts[2];
+        }
+
+        let regExp = new RegExp(regex, options);
```

Comment:
`regExp` should be build only once, and then used again and again. The same thing, build regExp in `filterGetType`

Reply:
Done.